4 - AnimalShelter/Code/AnimalShelter.cs

```diff
@@ -77,5 +81,16 @@ public OperationResult GetAnimalFromID(int uniqueAnimalId)
                 return new OperationResult("Multiple ID matches found - Shelter is in error state");
             }
         }
+        protected virtual bool IsSupported(AnimalType type)
+        {
+            var result =
+                (  type == AnimalType.Bird
+                || type == AnimalType.Cat
+                || type == AnimalType.Dog
+                || type == AnimalType.Snake);
+
+            return result;
+        }
```

Comment:
Nicely implemented! I could actually see this being public as well, so code can say "can this animal shelter support me", even if its not in the requirements.